enterprise/frontend/src/embedding-sdk/cli/steps/create-models-and-xrays.ts

```diff
@@ -43,9 +48,16 @@ export const createModelsAndXrays: CliStepMethod = async state => {
       dashboards.push({ id: dashboardId, name: model.modelName });
     }
 
+    // Populate the table metadata into the state (e.g. table fields),
+    // so we can use it for the permissions step.
+    const tablesWithMetadata = models.map(model => model.tableWithMetadata);
```

Comment:
We already query the metadata as part of `createModelFromTable`, so we might as well use the queried table with metadata to reduce number of queries needed.